test/Knp/Snappy/AbstractGeneratorTest.php

```diff
@@ -923,4 +923,26 @@ public function testleanupTemporaryFiles()
         $remove->setAccessible(true);
         $remove->invoke($generator);
     }
+
+    public function testResetOptions()
+    {
+        $media = $this->getMockBuilder(AbstractGenerator::class)
+            ->setMethods([
+                'resetOptions',
+                'setOption',
+            ])
+            ->setConstructorArgs(['the_binary', [], ['PATH' => '/usr/bin']])
+            ->getMock()
+        ;
+
+        $media->setOption('copies', 4);
+        $media->setOption('disable-javascript', true);
+        $media->setOption('no-background', true);
+
+        $media->resetOptions();
```

Comment:
Can you assert that options has been set before calling `resetOptions`?

Reply:
Setting option is already tested with `testSetOption()` method. I don't know we should test it here.